nucypher/characters/chaotic.py

```diff
@@ -264,27 +266,59 @@ def __repr__(self):
         #
         # REST Routes
         #
+        @rest_app.route("/status", methods=['GET'])
+        def status():
+            with ThreadedSession(self.db_engine) as session:
+                total_recipients = session.query(self.Recipient).count()
+                last_recipient = session.query(self.Recipient).filter(
+                    self.Recipient.last_disbursement_time.isnot(None)
+                ).order_by('last_disbursement_time').first()
+
+                last_address = last_recipient.address if last_recipient else None
+                last_transaction_date = last_recipient.last_disbursement_time.isoformat() if last_recipient else None
+
+                unfunded = session.query(self.Recipient).filter(
+                    self.Recipient.last_disbursement_time.is_(None)).count()
+
+                return json.dumps(
+                        {
+                            "total_recipients": total_recipients,
+                            "latest_recipient": last_address,
+                            "latest_disburse_date": last_transaction_date,
+                            "unfunded_recipients": unfunded,
+                            "state": {
+                                "eth": str(self.eth_balance),
+                                "NU": str(self.token_balance),
+                                "address": self.checksum_address,
+                                "contract_address": self.token_agent.contract_address,
+                            }
+                        }
+                    )
 
         @rest_app.route("/", methods=['GET'])
-        @limiter.limit("100/day;20/hour;1/minute")
         def home():
             rendering = render_template(self.TEMPLATE_NAME)
             return rendering
 
         @rest_app.route("/register", methods=['POST'])
-        @limiter.limit("5 per day")
         def register():
             """Handle new recipient registration via POST request."""
-            try:
-                new_address = request.form['address']
-            except KeyError:
-                return Response(status=400)  # TODO
 
-            if not eth_utils.is_checksum_address(new_address):
-                return Response(status=400)  # TODO
+            new_address = (
+                request.form.get('address') or
+                request.get_json().get('address')
+            )
+
+            if not new_address:
+                return Response(response="no address was supplied", status=411)
+
+            if not eth_utils.is_address(new_address):
+                return Response(response="an invalid ethereum address was supplied.  please ensure the address is a proper checksum.", status=400)
+            else:
+                new_address = eth_utils.to_checksum_address(new_address)
```

Comment:
:+1: 
We need to use this everywhere in the main codebase where we are case-sensitive to the address!